test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,294 @@
+    common_modules.ModuleInfo(
+        SingleAddModel, module_inputs_func=module_inputs_single_add
+    ),
+]
+
+# END OF SECTION TO MODIFY to add or modify nn.Module test cases ###############
+
+
+class _TestFxFrontendConsistency(common_utils.TestCase):
+    frontend_info: FrontendInfo
+
+    @common_modules.modules(
+        MODULES_DB, train_eval_mode=common_modules.TrainEvalMode.eval_only
+    )
+    def test_output_match(
+        self,
+        device: str,
+        dtype: torch.dtype,
+        module_info: common_modules.ModuleInfo,
+        training: bool,
+    ):
+        """Base test method for testing each module_info with fx frontend."""
+        # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+        assert device == "cpu"
+
+        if skip_reason := self.frontend_info.skips.get(module_info.formatted_name):
+            pytest.skip(skip_reason)
+
+        fx_frontend = self.frontend_info.fx_frontend
+
+        module_cls = module_info.module_cls
+        module_inputs = module_info.module_inputs_func(
+            module_info,
+            device=device,
+            dtype=dtype,
+            requires_grad=False,
+            training=training,
+        )
+
+        for (i, module_input) in enumerate(module_inputs):
+            with self.subTest(
+                module_name=module_info.formatted_name,
+                sample_num=i,
+                fx_frontend=fx_frontend.name,
+            ):
+                init_args, init_kwargs = (
+                    module_input.constructor_input.args,
+                    module_input.constructor_input.kwargs,
+                )
+                model = module_cls(*init_args, **init_kwargs)
+                model.to(device).to(dtype)
+                model.train(training)
+
+                args, kwargs = (
+                    module_input.forward_input.args,
+                    module_input.forward_input.kwargs,
+                )
+
+                if isinstance(fx_frontend, frontend.FxFrontendUnpackKwargs):
+                    graph_module, new_args = fx_frontend.trace(model, *args, **kwargs)
+                    fx_outputs = graph_module(*new_args)
+                else:
+                    graph_module = fx_frontend.trace(model, *args, **kwargs)
+                    fx_outputs = graph_module(*args, **kwargs)
+
+                outputs = model(*args, **kwargs)
+
+                self.assertEqual(
+                    pytree.tree_flatten(outputs)[0],
+                    pytree.tree_flatten(fx_outputs)[0],
+                )
+
+
+# 'Parameterize' the test class to run for each fx frontend.
+# Adopted from `parameterized` package.
+# The reason not using `parameterized` package is that it doesn't support
+# applying additional `common_device_type.instantiate_device_type_tests`.
+for frontend_info in FX_FRONTEND_INFOS:
+    test_class_module = sys.modules[_TestFxFrontendConsistency.__module__].__dict__
```

Comment:
I improved the parameterization in https://github.com/pytorch/pytorch/pull/92087/files#diff-e968c9cb6fc6631cab526cb3a9fe66358c4c6e757e2a223a224b976471bcb753R403-R412. May worth taking a look

Reply:
`parameterization` is not a good idea in Python `unittest`. It increases debug cost and reduces readability without providing actual benefits. If there is something you want to parameterize, you can use those parameters as function arguments.

Reply:
In theory we could parameterize the individual test methods. Is that what you mean by using parameters as function arguments? It’s harder to make it work here because PyTorch already has their own way of parameterizing OpInfos. We _could_ also extend PyTorch’s test mechanism with additional options we use, but that can mean effort. 

Can you say more on why it increases debug cost?